src/markdown/createMarkdownFile.ts

```diff
@@ -0,0 +1,69 @@
+import remarkParse from "remark-parse";
+import { unified } from "unified";
+import { visit } from "unist-util-visit";
+import { VFile } from "vfile";
+import { location } from "vfile-location";
+
+import { LanguageFileDefinition } from "../types/languages.js";
+import { NormalizedRuleReport, RuleReport } from "../types/reports.js";
+
+// Eventually, it might make sense to use a native speed Markdown parser...
+// However, the remark ecosystem is quite extensive and well-supported.
+// It'll be a while before we can replace it with a native parser.
+export function createMarkdownFile(
+	filePathAbsolute: string,
+	sourceText: string,
+): LanguageFileDefinition {
+	const virtualFile = new VFile({
+		path: filePathAbsolute,
+		value: sourceText,
+	});
+	const fileLocation = location(virtualFile);
+	const root = unified().use(remarkParse).parse(virtualFile);
+
+	return {
+		runRule(rule, options) {
+			const reports: NormalizedRuleReport[] = [];
+
+			const context = {
+				report: (report: RuleReport) => {
+					// We can assume these always exist.
+					/* eslint-disable @typescript-eslint/no-non-null-assertion */
+					const positionBegin = fileLocation.toPoint(report.range.begin)!;
+					const positionEnd = fileLocation.toPoint(report.range.end)!;
+					/* eslint-enable @typescript-eslint/no-non-null-assertion */
```

Comment:
This repeated col/line <> offset conversion logic is inconvenient. https://github.com/JoshuaKGoldberg/flint/issues/97